src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -57,8 +62,49 @@
     @ReceiveEvent(components = {SetTeamOnActivateComponent.class})
     public void onActivate(ActivateEvent event, EntityRef entity) {
         EntityRef player = event.getInstigator();
-        String team = setPlayerTeamToTeleporterTeam(player, entity);
-        handlePlayerTeleport(player, team);
+        if (isProperTeamSize(entity, player)) {
+            String team = setPlayerTeamToTeleporterTeam(player, entity);
+            handlePlayerTeleport(player, team);
+        }
+    }
+
+    private boolean isProperTeamSize(EntityRef teleporter, EntityRef player) {
+        setGamePrefab();
+        int oppositeTeamCount = 0;
+        int teleporterTeamCount = 0;
+        int maxTeamSizeDifference = gameEntity.getComponent(LASConfigComponent.class).maxTeamSizeDifference;
+        String teleporterTeam = teleporter.getComponent(LASTeamComponent.class).team;
+        Iterable<EntityRef> characters = entityManager.getEntitiesWith(PlayerCharacterComponent.class,
+                LASTeamComponent.class);
+
+        for (EntityRef character : characters) {
+            String otherPlayerTeam = character.getComponent(LASTeamComponent.class).team;
+            if (teleporterTeam.equals(otherPlayerTeam)) {
+                teleporterTeamCount++;
+            } else if (!otherPlayerTeam.equals(LASUtils.WHITE_TEAM)) {
+                oppositeTeamCount++;
+            }
+        }
+        if (teleporterTeamCount - oppositeTeamCount < maxTeamSizeDifference) {
+            teleporterTeamCount++;
+            gameEntity.getComponent(GameStateComponent.class).redTeamCount = teleporterTeam.equals(LASUtils.RED_TEAM) ? teleporterTeamCount : oppositeTeamCount;
+            gameEntity.getComponent(GameStateComponent.class).blackTeamCount = teleporterTeam.equals(LASUtils.RED_TEAM) ? oppositeTeamCount : teleporterTeamCount;
+            return true;
+        } else {
+            gameEntity.getComponent(GameStateComponent.class).redTeamCount = teleporterTeam.equals(LASUtils.RED_TEAM) ? teleporterTeamCount : oppositeTeamCount;
+            gameEntity.getComponent(GameStateComponent.class).blackTeamCount = teleporterTeam.equals(LASUtils.RED_TEAM) ? oppositeTeamCount : teleporterTeamCount;
+            player.getOwner().send(new ChatMessageEvent("The " + teleporterTeam + " team has more players so please join the " + LASUtils.getOppositionTeam(teleporterTeam)
+                    + " team.", EntityRef.NULL));
+            return false;
+        }
+    }
+
+    private void setGamePrefab() {
+        if (entityManager.getCountOfEntitiesWith(LASConfigComponent.class) == 0) {
+            gameEntity = entityManager.create("LightAndShadowResources:gameEntity");
+        } else {
+            gameEntity = entityManager.getEntitiesWith(LASConfigComponent.class).iterator().next();
+        }
```

Comment:
I don't understand why we need this :thinking:

Reply:
This isn't needed right now, but in the future if the gameEntity is created elsewhere for configuration or game state this would be needed then.

Reply:
Then I'd say, reintroduce it when it's actually needed.
We have a history of introducing things that might be useful or needed later on but never avtually getting to that point and thus cluttering our codebase.
Btw, that's not to say I don't believe in you getting to that point, just that with that track record we learned it's better to not do it irrespective of personal beliefs and hopes.